### ISSUE TITLE
Windows incremental build failure after only flamegpu2 project modified

### ISSUE THREAD

**Issue by Robadob** [FLAMEGPU/FLAMEGPU2#483]
We've had this issue for a long time. Rebuild an example, resave any `.cu` file in the main flamegpu2 library and then incrementally build the same example. It fails with a link error as `-dlink` has not been called for the new flamegpu2 library changes.

Pre CUDA 11.2, the only solution is to resave a file in the example and incrementally build. This tricks `-dlink` into occuring.

CUDA 11.2 It becomes possible to set the `CUDA Linker->Additional Dependencies` to point to the path of `flamegpu2.lib`. However it ii not currently possible to target this setting with CMake. ([CMake issue](https://gitlab.kitware.com/cmake/cmake/-/issues/21992)).

I've discussed this issue through with an NV QA team member, and it appears to be an issue with NVCC, as the `CUDA Linker->Use Host Library Dependencies` option is enabled, but host dependencies are not being detected for changes. I'm yet to receive confirmation that will be/has been fixed.

At this time it's not something we can automatically/permanently fix, but we have workaround to save time, so I will keep this updated as more progress is made.

**Comment by Robadob** [comment]
I've developed a fix for CUDA 11.2's visual studio integration.

Copy the file from inside this `.zip` to `C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Microsoft\VC\v160\BuildCustomizations`. You may wish to rename the existing file `CUDA 11.2.targets` to back it up first.

This file has been changed to add `$(ProjectDir)` to a variable defined at `CUDA 11.2.targets:398`. This is an MSBuild symbol for the directory containing `.vcxproj`, the same directory the host libraries are relative to (I would assume this is the working directory when MSBuild executes). As such the resolvepaths method used slightly below is able to resolve their paths (it otherwise drops the ones it can't resolve relative to specified directories).

[CUDA 11.2.targets.zip](https://github.com/FLAMEGPU/FLAMEGPU2/files/6213680/CUDA.11.2.targets.zip)

**Comment by ptheywood** [comment]
This can be worked around via cmake for CUDA 11.1+ (PR incoming), by additionally specifying the absolute path to the .lib file as a link library for the executable. 

CUDA 11.0 appears to not like the same lib file being provided multiple times with different paths, although this has not been directly tested on the PR. 

i.e. (with some extra logic to make it msvc/cuda version appropraite)
```
target_link_libraries(${NAME} "${CMAKE_CURRENT_BINARY_DIR}/$<TARGET_FILE:flamegpu2>")
```
